src/storage/src/source/kafka.rs

```diff
@@ -287,6 +300,51 @@ impl SourceReader for KafkaSourceReader {
     }
 }
 
+#[async_trait::async_trait]
+impl OffsetCommitter for KafkaOffsetCommiter {
+    async fn commit_offsets(
+        &self,
+        offsets: HashMap<PartitionId, MzOffset>,
+    ) -> Result<(), anyhow::Error> {
+        use rdkafka::consumer::CommitMode;
+        use rdkafka::topic_partition_list::Offset;
+
+        let mut tpl = TopicPartitionList::new();
+        for (pid, offset) in offsets.clone() {
+            // Note that we expect the above layers to pre-filter
+            // by partition for us. This is part of the
+            // `OffsetCommitter` contract.
+            let pid = match pid {
+                PartitionId::Kafka(id) => id,
+                _ => panic!("unexpected partition id type"),
+            };
+
+            // This matches the behavior of auto-commit, where we commit a
+            // pseudo-_frontier_. Additionally, overflow will be caught
```

Comment:
Why _pseudo-frontier_? This now behaves like the timely frontiers that we know and love, right?

Also: is it true that we jump through hoops to _subtract_ 1 in coming up with the source commit frontier and then _add_ 1 here again? :sweat_smile:

Reply:
If all sources turn out to work with these timely-style frontiers we might refactor that and get rid of the +1/-1 dance.

Reply:
yeah see this: https://materializeinc.slack.com/archives/CM7ATT65S/p1664232330711029

pg and kafka I think KINDOF use this frontier dance (kafka appears to by convention for committing offsets, but not for messages?